app/keepers.go

```diff
@@ -163,6 +165,46 @@ func InitStakingKeeper(appCodec codec.Codec, keys map[string]*sdk.KVStoreKey, ke
 	return &stakingK
 }
 
+// expandPath resolves paths containing the "~" character to the user's home directory.
+func expandPath(path string) string {
+	if !strings.HasPrefix(path, "~/") {
+		return path
+	}
+
+	homeDir := funcs.Must(os.UserHomeDir())
+
+	return filepath.Join(homeDir, path[2:])
+}
+
+func migrateWasmDir(oldWasmDir, newWasmDir string) error {
+	// Resolve paths
+	oldWasmPath, err := filepath.Abs(expandPath(oldWasmDir))
+	if err != nil {
+		return fmt.Errorf("failed to resolve absolute path for old wasm dir %s: %v", oldWasmDir, err)
+	}
+	newWasmPath, err := filepath.Abs(expandPath(newWasmDir))
+	if err != nil {
+		return fmt.Errorf("failed to resolve absolute path for new wasm dir %s: %v", newWasmDir, err)
+	}
```

Comment:
this should not be done in here but much earlier. E.g. we already know by construction that the old path is absolute

Reply:
wasm keeper does it's own check on `wasmDir` but I don't mind making our check more global if it's not set to the default `""`